tests/v2/validation/KDM/kdm_test.go

```diff
@@ -0,0 +1,228 @@
+
+	result, err := request.Raw()
+	k.Require().NoError(err, "error executing command")
+	return string(result)
+}
+
+func findLatestVersion(allVersions []string) (string, error) {
+	var latest *semver.Version
+	for _, version := range allVersions {
+		ver, err := semver.NewVersion(version)
+		if err != nil {
+			return "", err
+		}
+		if latest == nil || ver.GreaterThan(latest) {
+			latest = ver
+		}
+	}
+	if latest == nil {
+		return "", fmt.Errorf("no valid versions found")
+	}
+	return latest.String(), nil
+}
+
+func (k *KDMTestSuite) TestChangeKDMurl() {
+	// change kdm url to release
+	k.updateKDMurl("https://raw.githubusercontent.com/vatsalparekh/rancher/kdm-test-dummy/data.json")
+
+	// scale Rancher to 3 replicas
+	k.ScaleRancherTo(3)
+
+	// get the current release value
+	availableRKE2Versions, err := kubernetesversions.ListRKE2AllVersions(k.client)
+	k.Require().NoError(err, "error listing RKE2 versions")
+	currentRKE2Version, err := findLatestVersion(availableRKE2Versions)
+	k.Require().NoError(err, "error getting kubernetes version")
+	log.Infof("current RKE2 version available: %s", currentRKE2Version)
+
+	// change kdm url to dev
+	k.updateKDMurl("https://releases.rancher.com/kontainer-driver-metadata/dev-v2.8/data.json")
+	// arbitrary sleep to let all pods update the KDM values in channelserver
+	time.Sleep(30 * time.Second)
+
+	var updatedRKE2Version string
+	// check latest Release value
+	err = wait.ExponentialBackoff(defaultBackoff, func() (bool, error) {
+		currentAvailableRKE2Versions, err := kubernetesversions.ListRKE2AllVersions(k.client)
+		if err != nil {
+			return false, fmt.Errorf("error getting kubernetes version: %s", err.Error())
+		}
+		updatedRKE2Version, err = findLatestVersion(currentAvailableRKE2Versions)
+		if updatedRKE2Version != currentRKE2Version {
+			// change detected
+			return true, nil
+		}
+		return false, nil
+	})
+	log.Infof("New available RKE2 version: %s", updatedRKE2Version)
+	if updatedRKE2Version != currentRKE2Version {
+		// look for updated version in all Rancher Pod
+		cmd := []string{"curl", "--insecure", "https://0.0.0.0/v1-rke2-release/releases"}
+		pods := k.GetRancherReplicas()
+		for _, pod := range pods.Items {
+			output := k.ExecCMDForKDMDump(pod, cmd)
+			// if the curl output to from the pod to itslef (over 0.0.0.0) doesn't contain (trace of) version we
+			// recieved over API call which most likely went to the leader Pod, we can assume the KDM file was
+			// NOT updated on all Pods
+			if !strings.Contains(output, updatedRKE2Version) {
+				k.Fail(fmt.Sprintf("found KDM from a pod:%v not having the latest known version:%v", pod.Name, updatedRKE2Version))
+			}
+		}
+	} else {
+		// This is the scenario where both release and dev version of KDM have same latest version
+		log.Infof("Latest RKE2 versions on both release and dev KDM channels seem to be the same, %v & %v", currentRKE2Version, updatedRKE2Version)
+	}
+}
+
+func TestKDM(t *testing.T) {
```

Comment:
One more nit 😅 -- i'm not sure this documented anywhere, but its our convention to have `TestSuite` as a suffix to these functions which kick off the suite -- could you update this function name to `TestKDMTestSuite`?